lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -520,10 +520,9 @@ bool BorrowedValue::visitInteriorPointerOperands(
 //                           InteriorPointerOperand
 //===----------------------------------------------------------------------===//
 
-bool InteriorPointerOperand::getImplicitUses(
-    SmallVectorImpl<Operand *> &foundUses,
+bool InteriorPointerOperand::getImplicitUsesForAddress(
+    SILValue projectedAddress, SmallVectorImpl<Operand *> &foundUses,
```

Comment:
I'm not sure what this method is doing or what an "implicit use" is. Think it's looking at the transitive uses of address to find the lifetime extent of the interior pointer? Maybe it should be called something like `findTransitiveInteriorPointerUses` with a comment to explain that it defines the maximum extent of the interior pointer's lifetime, which must be within the root object's borrow scope.

Reply:
I'm also not sure why it's recording all uses even those that are transitively followed

Reply:
You are right. Let me fix that. This is basically the way that the optimizer takes an address and finds all of its uses exhaustively. I think it should be rewritten based off of the AccessUseDefChainVisitor IMO.